ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/MachineTestsTable.test.tsx

```diff
@@ -275,4 +288,134 @@ describe("MachineTestsTable", () => {
       wrapper.find("Button[data-test='action-menu-show-metrics']").exists()
     ).toEqual(false);
   });
+
+  it("sends an analytics event when suppressing a script result", () => {
+    state.nodescriptresult.items = { abc123: [1] };
+    const scriptResults = [
+      scriptResultFactory({
+        id: 1,
+        result_type: ResultType.Testing,
+        status: scriptStatus.FAILED,
+        suppressed: false,
+      }),
+    ];
+    state.scriptresult.items = scriptResults;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <MachineTestsTable machineId="abc123" scriptResults={scriptResults} />
+        </MemoryRouter>
+      </Provider>
+    );
+
+    const checkbox = wrapper.find('input[data-test="suppress-script-results"]');
+    checkbox.simulate("change", { target: { value: "checked" } });
```

Comment:
If the suppress action triggered on checked out is there a confirmation step?

Reply:
There is no confirmation step with the suppress checkboxes. A websocket call is made immediately when the checkbox is checked.